File: sami/step_definitions/Multiple Shipping Addresses.js
import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps';
import CheckoutPage from '../page-objects/CheckoutPage';

const checkoutPage = new CheckoutPage();

Given('the user is on the checkout page', () => {
  checkoutPage.visitCheckoutPage();
});

Given('the user has items in the cart', () => {
  checkoutPage.verifyItemsInCart();
});

When('the user chooses to ship to multiple addresses', () => {
  checkoutPage.selectMultipleShippingAddresses();
});

Then('the user should see an option to select different addresses for each item', () => {
  checkoutPage.verifyMultipleAddressOption();
});

When('the user adds a new shipping address', () => {
  checkoutPage.addNewShippingAddress();
});

Then('the new address should be available for selection in the address list', () => {
  checkoutPage.verifyNewAddressInList();
});

Given('the user has multiple addresses available', () => {
  checkoutPage.verifyMultipleAddressesAvailable();
});

When('the user assigns different addresses to each item in the order', () => {
  checkoutPage.assignAddressesToItems();
});

Then('each item should display the assigned address in the order summary', () => {
  checkoutPage.verifyAssignedAddressesInSummary();
});

Given('the user has assigned different addresses to items', () => {
  checkoutPage.verifyAssignedAddresses();
});

When('the user reviews the order summary', () => {
  checkoutPage.reviewOrderSummary();
});

Then('the order summary should display each item with its respective shipping address', () => {
  checkoutPage.verifyOrderSummaryAddresses();
});

Given('the user has reviewed the order summary', () => {
  checkoutPage.verifyOrderSummaryReviewed();
});

When('the user confirms the order', () => {
  checkoutPage.confirmOrder();
});

Then('the order should be processed with items being shipped to their respective addresses', () => {
  checkoutPage.verifyOrderProcessed();
});

Then('the user should receive a confirmation email with details of each shipment', () => {
  checkoutPage.verifyConfirmationEmail();
});

File: step_definitions/Multiple Shipping Addresses.java
package com.cucumber.steps;

import com.framework.cucumber.TestHarness;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

public class MultipleShippingAddressesSteps extends MultipleShippingAddressesPage {

    private TestHarness testHarness = new TestHarness();

    @Given("the user is on the checkout page")
    public void theUserIsOnTheCheckoutPage() {
        navigateToCheckoutPage();
    }

    @When("the user selects the option to ship items to multiple addresses")
    public void theUserSelectsTheOptionToShipItemsToMultipleAddresses() {
        selectMultipleShippingAddressesOption();
    }

    @Then("the user should be presented with a list of items in the order")
    public void theUserShouldBePresentedWithAListOfItemsInTheOrder() {
        verifyItemsListDisplayed();
    }

    @Then("each item should have an option to select a shipping address")
    public void eachItemShouldHaveAnOptionToSelectAShippingAddress() {
        verifyShippingAddressOptionForItems();
    }

    @Given("the user has selected multiple shipping addresses option")
    public void theUserHasSelectedMultipleShippingAddressesOption() {
        verifyMultipleShippingAddressesOptionSelected();
    }

    @When("the user assigns different addresses to each item")
    public void theUserAssignsDifferentAddressesToEachItem() {
        assignDifferentAddressesToItems();
    }

    @Then("each item should display the selected shipping address")
    public void eachItemShouldDisplayTheSelectedShippingAddress() {
        verifySelectedShippingAddressesForItems();
    }

    @Given("the user has assigned different shipping addresses to items")
    public void theUserHasAssignedDifferentShippingAddressesToItems() {
        verifyDifferentShippingAddressesAssigned();
    }

    @When("the user confirms the shipping address selection")
    public void theUserConfirmsTheShippingAddressSelection() {
        confirmShippingAddressSelection();
    }

    @Then("the system should save the shipping address for each item")
    public void theSystemShouldSaveTheShippingAddressForEachItem() {
        verifyShippingAddressesSaved();
    }

    @Then("the order summary should reflect the different shipping addresses")
    public void theOrderSummaryShouldReflectTheDifferentShippingAddresses() {
        verifyOrderSummaryReflectsShippingAddresses();
    }

    @Given("the user has saved the shipping address selection")
    public void theUserHasSavedTheShippingAddressSelection() {
        verifyShippingAddressSelectionSaved();
    }

    @When("the user chooses to edit the shipping addresses")
    public void theUserChoosesToEditTheShippingAddresses() {
        editShippingAddresses();
    }

    @Then("the user should be able to change the shipping addresses for items")
    public void theUserShouldBeAbleToChangeTheShippingAddressesForItems() {
        verifyShippingAddressesCanBeChanged();
    }

    @Then("the changes should be reflected in the order summary")
    public void theChangesShouldBeReflectedInTheOrderSummary() {
        verifyOrderSummaryReflectsChanges();
    }

    @Given("the user has finalized the shipping addresses")
    public void theUserHasFinalizedTheShippingAddresses() {
        verifyShippingAddressesFinalized();
    }

    @When("the user completes the order process")
    public void theUserCompletesTheOrderProcess() {
        completeOrderProcess();
    }

    @Then("the order confirmation should list all items with their respective shipping addresses")
    public void theOrderConfirmationShouldListAllItemsWithTheirRespectiveShippingAddresses() {
        verifyOrderConfirmationListsShippingAddresses();
    }

    @Then("the user should receive a confirmation email with the shipping details for each item")
    public void theUserShouldReceiveAConfirmationEmailWithTheShippingDetailsForEachItem() {
        verifyConfirmationEmailReceived();
    }
}